Apollo client being initialized twice

I see the following on the Apollo 2.0.7 demo:

![screen shot 2017-08-31 at 2 28 50 pm](https://user-images.githubusercontent.com/4530272/29941654-e3281eb0-8e58-11e7-92ac-db9c23089672.png)

Note how there are redundant options in the View menu.

The console gives the following error:
```
Browser.js:1981 Error: Tried to register widget with id==menubar_powered_by_jbrowse but that id is already registered Error: Tried to register widget with id==menubar_powered_by_jbrowse but that id is already registered
    at Object.add (http://demo.genomearchitect.org/Apollo-staging/4126658216003575441700795384/jbrowse/src/dijit/registry.js:26:11)
    at Object.create (http://demo.genomearchitect.org/Apollo-staging/4126658216003575441700795384/jbrowse/src/dijit/_WidgetBase.js:428:13)
    at Object.postscript (http://demo.genomearchitect.org/Apollo-staging/4126658216003575441700795384/jbrowse/src/dijit/_WidgetBase.js:366:9)
    at new <anonymous> (http://demo.genomearchitect.org/Apollo-staging/4126658216003575441700795384/jbrowse/src/dojo/_base/declare.js:406:7)
    at Object.postCreateMenus (http://demo.genomearchitect.org/Apollo-staging/4126658216003575441700795384/jbrowse/plugins/WebApollo/js/main.js:718:33)
    at http://demo.genomearchitect.org/Apollo-staging/4126658216003575441700795384/jbrowse/plugins/WebApollo/js/main.js:222:23
    at http://demo.genomearchitect.org/Apollo-staging/4126658216003575441700795384/jbrowse/src/JBrowse/Browser.js:1979:29
    at signalListener (http://demo.genomearchitect.org/Apollo-staging/4126658216003575441700795384/jbrowse/src/dojo/Deferred.js:37:21)
    at signalWaiting (http://demo.genomearchitect.org/Apollo-staging/4126658216003575441700795384/jbrowse/src/dojo/Deferred.js:28:4)
    at Deferred.resolve (http://demo.genomearchitect.org/Apollo-staging/4126658216003575441700795384/jbrowse/src/dojo/Deferred.js:192:5) Error: Tried to register widget with id==menubar_powered_by_jbrowse but that id is already registered
    at Object.add (registry.js:26)
    at Object.create (_WidgetBase.js:428)
    at Object.postscript (_WidgetBase.js:366)
    at new <anonymous> (declare.js:406)
    at Object.postCreateMenus (main.js:718)
    at main.js:222
    at Browser.js:1979
    at signalListener (Deferred.js:37)
    at signalWaiting (Deferred.js:28)
    at Deferred.resolve (Deferred.js:192)
```

--- a/grails-app/controllers/org/bbop/apollo/JbrowseController.groovy
+++ b/grails-app/controllers/org/bbop/apollo/JbrowseController.groovy
@@ -460,6 +460,8 @@ class JbrowseController {
             }
         }
 
+        removeIncludedPlugins(jsonObject.plugins)
+
         // add extendedTrackList.json, if available
         if (!currentOrganism.dataAddedViaWebServices) {
             println "${configWrapperService.commonDataDirectory + File.separator + currentOrganism.id + "-" + currentOrganism.commonName + File.separator + OrganismController.EXTENDED_TRACKLIST}"
@@ -473,6 +475,26 @@ class JbrowseController {
 
         response.outputStream << jsonObject.toString()
         response.outputStream.close()
+    }
+
+    /**
+     * Removes plugins included in annot.json (which is just WebApollo)
+     * @param pluginsArray
+     */
+    def removeIncludedPlugins(JSONArray pluginsArray) {
+        for(plugin in pluginsArray){
+            if(plugin instanceof JSONObject){
+                if(plugin.name == "WebApollo"){
+                    pluginsArray.remove(plugin)
+                }
+            }
+            else
+            if(plugin instanceof String){
+                if(plugin=="WebApollo"){
+                    pluginsArray.remove(plugin)
+                }
+            }
+        }
     }
 
     private static boolean isCacheableFile(String fileName) {

--- a/sample-h2-apollo-config.groovy
+++ b/sample-h2-apollo-config.groovy
@@ -72,9 +72,6 @@ environments {
 //        alwaysRecheck = true
 //    }
 //    plugins {
-//        WebApollo{
-//            included = true
-//        }
 //        NeatHTMLFeatures{
 //            included = true
 //        }

--- a/sample-mysql-apollo-config.groovy
+++ b/sample-mysql-apollo-config.groovy
@@ -71,9 +71,6 @@ environments {
 //        alwaysRecheck = true
 //    }
 //    plugins {
-//        WebApollo{
-//            included = true
-//        }
 //        NeatHTMLFeatures{
 //            included = true
 //        }

--- a/sample-postgres-apollo-config.groovy
+++ b/sample-postgres-apollo-config.groovy
@@ -119,9 +119,6 @@ environments {
 //	// We advise caution if enabling these plugins with Apollo until this process is finalized.
 //    }
 //    plugins {
-//        WebApollo{
-//            included = true
-//        }
 //        NeatHTMLFeatures{
 //            included = true
 //        }

